src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/vsts_cd_provider.py

```diff
@@ -0,0 +1,62 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from __future__ import print_function
+from sys import stderr
+from vsts_cd_manager.continuous_delivery_manager import ContinuousDeliveryManager
+from azure.cli.core._profile import Profile
+
+
+class VstsContinuousDeliveryProvider(object):
+    def __init__(self):
+        self._progress_last_message = ''
+
+    # pylint: too-many-arguments
+    def setup_continuous_delivery(self,  # pylint:disable=too-many-arguments
+                                  resource_group_name, name, repo_url, branch, git_token,
+                                  slot, cd_app_type, cd_account, cd_create_account, location):
+        """
+        This method sets up CD for an Azure Web App thru Team Services
+        """
+
+        # Gather information about the Azure connection
+        profile = Profile()
+        subscription = profile.get_subscription()
+        user = profile.get_current_account_user()
+        cred, _, _ = profile.get_login_credentials(subscription_id=None)
+
+        cd_manager = ContinuousDeliveryManager(self._update_progress)
+
+        # Generate an Azure token with the VSTS resource app id
+        auth_token = profile.get_access_token_for_resource(user, None, cd_manager.get_vsts_app_id())
+
+        cd_manager.set_repository_info(repo_url, branch, git_token)
+        cd_manager.set_azure_web_info(resource_group_name, name, cred, subscription['id'],
+                                      subscription['name'], subscription['tenantId'], location)
+        vsts_cd_status = cd_manager.setup_continuous_delivery(slot, cd_app_type, cd_account,
+                                                              cd_create_account, auth_token)
+        return vsts_cd_status
+
+    # pylint: disable=no-self-use
+    def remove_continuous_delivery(self):
+        """
+        To be Implemented
+        """
+        # TODO: this would be called by appservice web source-control delete
+        pass
+
+    def _update_progress(self, current, total, status):
+        if total:
+            percent_done = current * 100 / total
```

Comment:
I'm not sure about the type of the current, but if it is integer this express results different results in python 2 and python 3:
```
✔ ~/code 
14:14 $ python -c "print(100/8)"
12
✔ ~/code 
14:14 $ python3 -c "print(100/8)"
12.5
```